### PR TITLE
fix: do not display undeployed applications

### PR DESCRIPTION
**Pull request by mauromorales**
How and when to display these applications needs to be defined

## Context

We are currently in the process of refactoring all application commands to deal with user applications and not deployments in general (relates to #6097). However there are some additional cases that might need to be addressed, one of them is **undeployed applications** e.g. you create a quickstart but for some reason deployment fails, you are left with one source repository that has no deployments. It would be good to have a way to display these in the get application command, either by default or through a flag.

I'm reverting this change because atm there's a BDD test that breaks with the following error:

```
reates a spring application and promotes it to staging
       [It]
      /workspace/source/test/suite/spring/jx_create_spring.go:36
      get applications with a URL
      Unexpected error:
          <*errors.fundamental | 0xc000517420>: {
              msg: "must be at least 3 fields in bdd-jx, entire output was APPLICATION           STAGING PODS URL\nbdd-jx                             \nbdd-spring-1575891571 0.0.1   1/1  http://bdd-spring-1575891571-jx-staging.cbjx-vs-pr-200-46arc.staging-jxaas.live",
              stack: [0x17fa1c4, 0x18aba3c, 0x1740360, 0x18abedc, 0xfd5589, 0x18a9504, 0x18a94dc, 0x18ac8ea, 0x1740360, 0x18a6e8c, 0x18acfc9, 0x18acfa6, 0x1740360, 0x18a7af4, 0x18b14a3, 0x1740360, 0x18b2029, 0x172fd2c, 0x172f99f, 0x172ee34, 0x1735e06, 0x17356b4, 0x173b5ff, 0x173b114, 0x173a957, 0x173cfbe, 0x173f997, 0x173f6
dd, 0x18a2782, 0x18b2a50, 0x504fa0, 0x45eaf1],
          }
          must be at least 3 fields in bdd-jx, entire output was APPLICATION           STAGING PODS URL
          bdd-jx
          bdd-spring-1575891571 0.0.1   1/1  http://bdd-spring-1575891571-jx-staging.cbjx-vs-pr-200-46arc.staging-jxaas.live
      occurred
```

Until the proper functionality is defined, we've decided to keep the functionality for undeployed applications untouched.

### CODE REVIEW

**Comment by mauromorales** [comment]
fixes #6308

File: pkg/cmd/get/get_applications.go [hunk 1]
```diff
@@ -131,12 +131,17 @@ func (o *GetApplicationsOptions) Run() error {
...
 
+OUTER:
 	for _, a := range list.Items {
 		row := []string{}
 		name := a.Name()
 
 		for _, k := range o.sortedKeys(list.Environments()) {
 
+			if len(a.Environments) == 0 {
+				continue OUTER
```
Comment (macox) [inline pkg/cmd/get/get_applications.go]: maybe just use a `break` here to reduce complexity

Comment (mauromorales) [reply pkg/cmd/get/get_applications.go]: @macox if I break that only affects the inner loop not the outer one, or am I missing something?

**Pull request merged**